dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/Config13DefaultSources.java

```diff
@@ -37,7 +47,19 @@
         sources.add(new ServerXMLDefaultVariableConfigSource());
 
         sources.addAll(getPropertiesFileConfigSources(classloader));
+        if (!listOfConfigSources.isEmpty())
+            sources.addAll(listOfConfigSources);
 
         return sources;
     }
+
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+    protected synchronized void setConfigSource(ConfigSource configSource) {
```

Comment:
Not sure about adding this reference in here and making the defaultsources implementation a component.

I agree we do want a new service that allows configsources to be provided by other features but I'm not sure that this is the best place or way to do it. (I'm not sure it should be in this class and I think it should probably be MULTIPLE instead of OPTIONAL)

Can you talk to @tevans78 tomorrow morning? He should be able to help you work out the best place to put this.

Reply:
Yes, I was thinking about adding some function similar to this but a) I wanted to talk to @Emily-Jiang first and b) due to the way the config bundles build together, any DS components need to be exposed through the `com.ibm.ws.microprofile.config.1.3.services` bundle and not this one.

Reply:
I created a separate prototype PR (we should keep it separate) ... https://github.com/OpenLiberty/open-liberty/pull/7682